### PR TITLE
ci: reflect to clap updates

### PR DESCRIPTION
**Pull request by weihanglo**
### What does this PR try to resolve?

CI is failing due to clap updating its error messages.

### Additional information

The update of `snapbox` tests is lovely. Just use `SNAPSHOTS=overwrite` and you're all set.

I wish other tests can do that as well.

### CODE REVIEW

**Review by epage — APPROVED** [review]
As for other tests, I want to add in-source snapshotting to snapbox but need a feature from cargo first (and am distracted by toml).

**Review by epage — COMMENTED** [review]
Can you update the clap requirement so local lockfiles are forced to update?

**Comment by weihanglo** [comment]
> Can you update the clap requirement so local lockfiles are forced to update?

Done. Sorry for forgetting doing that 😆

**Comment by ehuss** [comment]
Just beware that updating the requirement *might* incur some difficulties updating in rust-lang/rust. That introduces the `is-terminal` dependency (which in turn includes `rustix`). It may impact other tools using clap. I ran into some complications with clap 4 in https://github.com/rust-lang/rust/pull/106520.  Updating might get blocked by that PR.

**Comment by weihanglo** [comment]
To unblock Cargo's own CI pipeline, I suggest not updating Cargo.toml at this moment. Does it sound OK to you, @epage?

**Comment by epage** [comment]
If we don't update the lock file, the tests will fail depending on what version you are using.

If we can't upgrade the clap version at this time, it seems like what we really should be doing is committing a lock file.  Always testing the latest doesn't help the rust repo at all because we could be depending on features that aren't in their lockfile.  Always testing the latest also runs into these kinds of issues.

**Comment by ehuss** [comment]
I would suggest moving forward with this now (w/updating Cargo.toml), and deal with the upstream update over the next week. We can let Mark know that it is blocking cargo updates.

A shorter term workaround if that gets delayed is to update `jsondoclint`. I *think* that is currently the only tool that will cause problems with clap (I haven't checked, though).  Either we can remove the color feature from that, or remove clap from it altogether. It is a very trivial little utility, and I'm not sure it really needs something as big as clap to just get 3 args.

**Comment by weihanglo** [comment]
@bors retry

<https://github.com/rust-lang/cargo/actions/runs/3917884106/jobs/6697948436>

Probably <https://github.com/rust-lang/cargo/issues/11334>? Will investigate later then.

**Pull request merged**
